### ISSUE TITLE
variable not in scope with field selectors, GHC 9.8, OverloadedRecordDot

### ISSUE THREAD

**Issue by guibou** [guibou/PyF#133]
The following code:

```haskell
data Horse = Horse { turtle :: Int }

h = Horse 10

s = [fmt|{turtle h}|]
```

Would lead to a "variable not in scope" error for `turtle` when `OverloadedRecordDot` extension is enabled.

# Workaround

use `h.turtle` syntax.

# Reasons

Template haskell in GHC 9.8 put the record selectors in a different namespace and hence `lookupName` cannot resolve it anymore and PyF think the variable does not exists and raises an error.

I'm unsure if a definitive workaround exists.

See for details: https://gitlab.haskell.org/ghc/ghc/-/wikis/migration/9.8#new-namespace-for-record-fields

**Comment by guibou** [comment]
https://gitlab.haskell.org/ghc/ghc/-/issues/24293 upstream ticket

**Comment by guibou** [comment]
Fixed upstream in GHC 9.8.2.
